tests/test_ptc.py

```diff
@@ -95,6 +97,229 @@ def setUp(self):
+                # Test that the arrays "covs" (FFT) and "covDirect" (direct space) are the same
+                for row1, row2 in zip(covs, covsDirect):
+                    for a, b in zip(row1, row2):
+                        self.assertAlmostEqual(a, b)
+                tupleRows += [(mu1, mu2) + cov + (ampNumber, expTime, amp) for cov in covs]
+                tags = ['mu1', 'mu2', 'i', 'j', 'var', 'cov', 'npix', 'ext', 'expTime', 'ampName']
+            allTags += tags
+            tupleRecords += tupleRows
+        covariancesWithTags = np.core.records.fromrecords(tupleRecords, names=allTags)
+        covFits, _ = fitData(covariancesWithTags)
+        dataset = task.getOutputPtcDataCovAstier(covFits)
+
+        # Chek the gain and that the ratio of the variance caclulated via cov Astier (FFT) and
+        # that calculated with the standard PTC is close to 1.
+        for amp in self.ampNames:
+            self.assertAlmostEqual(dataset.gain[amp][0], 0.75, places=2)
+            for v1, v2 in zip(varStandard[amp], dataset.varAdu[amp][0]):
+                self.assertAlmostEqual(v1/v2, 1.0, places=1)
+
+    def getCovariancesAstierDirect(self, exposure1, exposure2, config, region=None):
+        """Calculate covariances of a difference image in real space"""
+        if region is not None:
+            im1Area = exposure1.maskedImage[region]
+            im2Area = exposure2.maskedImage[region]
+        else:
+            im1Area = exposure1.maskedImage
+            im2Area = exposure2.maskedImage
+
+        im1Area = afwMath.binImage(im1Area, config.binSize)
+        im2Area = afwMath.binImage(im2Area, config.binSize)
+
+        statsCtrl = afwMath.StatisticsControl()
+        statsCtrl.setNumSigmaClip(config.nSigmaClipPtc)
+        statsCtrl.setNumIter(config.nIterSigmaClipPtc)
+        #  Clipped mean of images; then average of mean.
+        mu1 = afwMath.makeStatistics(im1Area, afwMath.MEANCLIP, statsCtrl).getValue()
+        mu2 = afwMath.makeStatistics(im2Area, afwMath.MEANCLIP, statsCtrl).getValue()
+        mu = 0.5*(mu1 + mu2)
+
+        # Take difference of pairs
+        # symmetric formula: diff = (mu2*im1-mu1*im2)/(0.5*(mu1+mu2))
+        temp = im2Area.clone()
+        temp *= mu1
+        diffIm = im1Area.clone()
+        diffIm *= mu2
+        diffIm -= temp
+        diffIm /= mu
+
+        # Get the mask and identify good pixels as '1', and the rest as '0'.
+        w1 = np.where(im1Area.getMask().getArray() == 0, 1, 0)
+        w2 = np.where(im2Area.getMask().getArray() == 0, 1, 0)
+
+        w12 = w1*w2
+        wDiff = np.where(diffIm.getMask().getArray() == 0, 1, 0)
+        w = w12*wDiff
+
+        maxRangeCov = config.maximumRangeCovariancesAstier
+        covs = self.computeCovDirect(diffIm.getImage().getArray(), w, maxRangeCov)
+
+        return mu1, mu2, covs
+
+    def computeCovDirect(self, diffImage, weightImage, maxRange):
```

Comment:
Same with this.  If this is a way to calculate things, it should be in the ptc task as an optional way to calculate things.